src/main/java/seedu/address/model/tag/TagManagerImpl.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.model.tag;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.model.person.Person;
+
+/**
+ * Concrete implementation class of {@code TagManager} that uses a {@code HashMap}.
+ */
+public class TagManagerImpl implements TagManager {
+
+    private static final String MESSAGE_ERROR_PERSON_NOT_FOUND = "Person instance not found in tag manager!";
+
+    private final Map<Tag, Set<Person>> tagPersonSetMap;
+
+    public TagManagerImpl() {
+        tagPersonSetMap = new HashMap<>();
+    }
+
+    @Override
+    public Set<Person> getPersonsUnderTag(Tag tag) {
+        return tagPersonSetMap.get(tag) == null ? Set.of() : tagPersonSetMap.get(tag);
```

Comment:
could be wrapped in an `Optional` (since you're doing so below anyway)

Reply:
Is there a difference in wrapping in an Optional? It would still be 
`Optional.ofNullable(tagPersonSetMap.get(tag)).orElse(Set.of())` which I don't think is much better

Reply:
nope, just a personal preference.